main.c

```diff
@@ -15,14 +15,14 @@
 static void dump_memory_map()
 {
     printf("Memory map:\n");
-    for (int i = 0; i < lib_sysinfo.n_memranges; i++) {
+    for (size_t i = 0; i < lib_sysinfo.n_memranges; i++) {
```

Comment:
`n_memranges` is an `int`. So changing `i` to be `size_t` actually introduces an issue, because you are comparing signed and unsigned integers.

This should result in a compiler warning, does it not?

Reply:
why is that var an int? even its name prefix `n_` suggests that it is a natural number and these are unsigned.

Instead of propagating this questionable type choice i'd rather cast the limit variable to size_t.

However, feel free to ignore my comments as i am basically just a caring zaungast.